x-pack/plugins/observability_solution/dataset_quality/server/routes/data_streams/get_degraded_docs.ts

```diff
@@ -99,26 +101,27 @@ export async function getDegradedDocsPaginated(options: {
       aggs: aggs(after?.docsCount),
     },
   ]);
+  const [degradedDocsResponse, totalDocsResponse] = response.responses;
 
   const currDegradedDocs =
-    response.responses[0].aggregations?.datasets.buckets.map((bucket) => ({
+    degradedDocsResponse.aggregations?.datasets.buckets.map((bucket) => ({
       dataset: `${type}-${bucket.key.dataset}-${bucket.key.namespace}`,
       count: bucket.doc_count,
     })) ?? [];
 
   const degradedDocs = [...prevResults.degradedDocs, ...currDegradedDocs];
 
   const currTotalDocs =
-    response.responses[1].aggregations?.datasets.buckets.map((bucket) => ({
+    totalDocsResponse.aggregations?.datasets.buckets.map((bucket) => ({
       dataset: `${type}-${bucket.key.dataset}-${bucket.key.namespace}`,
       count: bucket.doc_count,
     })) ?? [];
 
   const docsCount = [...prevResults.docsCount, ...currTotalDocs];
 
   if (
-    response.responses[0].aggregations?.datasets.after_key ||
-    response.responses[1].aggregations?.datasets.after_key
+    totalDocsResponse.aggregations?.datasets.after_key &&
```

Comment:
Why do we check only for `totalDocsResponse`? are we 100% sure that `degradedDocsResponse` wouldn't overflow the request (having more than the `SIZE_LIMIT` if totalDocsResponse is smaller than the `SIZE_LIMIT`?

Reply:
`totalDocsResponse` should always be larger in count than `degradedDocsResponse` because the latter uses the same query plus an additional filter.